react/components/RenderProvider.tsx

```diff
@@ -944,6 +928,61 @@ class RenderProvider extends Component<Props, RenderProviderState> {
       }
     )
   }
+
+  private prefetchPages = () => {
+    const { prefetchRoutes } = this.state
+
+    if (prefetchRoutes.size >= 0) {
+      window.requestIdleCallback
+        ? window.requestIdleCallback(this.execPrefetchPages)
+        : setTimeout(this.execPrefetchPages, 3000)
+    }
+  }
+
+  private execPrefetchPages = async () => {
+    const {
+      runtime,
+      runtime: { renderMajor },
+    } = this.props
+
+    const {
+      pages,
+      prefetchRoutes,
+      culture: { locale },
+    } = this.state
+
+    const {
+      components: defaultComponents,
+      extensions: defaultExtensions,
+      messages: defaultMessages,
+    } = await fetchDefaultPages({
+      apolloClient: this.apolloClient,
+      locale,
+      pages,
+      renderMajor,
+      routeIds: Array.from(prefetchRoutes),
```

Comment:
Better clean up prefetchRoutes

```ts
const routeIds = Array.from(prefetchRoutes)
prefetchRoutes.clear()
```

Reply:
prefetchRoutes is only read once, its nice to have it's data in case we need debugging.